crates/maybenot/src/framework.rs

```diff
@@ -174,8 +174,14 @@ where
     }
 
     /// Trigger zero or more [`TriggerEvent`] for all machines running in the
-    /// framework. The current time SHOULD be the current time at time of
+    /// framework.
+    ///
+    /// The `current_time` SHOULD be the current time at time of
     /// calling the method (e.g., [`Instant::now()`](std::time::Instant::now())).
+    /// The `current_time` must be a monotonically nondecreasing clock. This means that the time
+    /// passed in must never be earlier than what was given to [`Framework::new()`] or a previous
+    /// call to `trigger_event` for the same framework instance.
```

Comment:
To clarify: how does this interact with the warnings on https://doc.rust-lang.org/nightly/std/time/struct.Instant.html#monotonicity that `std::time::Instant` can be non-monotonic under some circumstances?  If this code is running in a VM or on dubious hardware, does the caller need to manually insure that Instant is monotonic?

(As an alternative, maybe we could clarify _how_ bad it is if the clock skips backwards?  In other words, does it panic, or will it give incorrect results, or something else?)

Reply:
I'll leave this to @pylls to answer.

I'd argue we have not made anything *worse*. The API is identical to before this PR, just that it allows more clock sources. For the sake of making progress I suggest merging this as is and anyone is then free to submit documentation updates as they see fit?

Reply:
Thanks again @faern and @nmathewson ! I'll do as suggested: merge this now and add some extra documentation in a later commit to `v2`.